### ISSUE TITLE
Testing options for 2.5.1 Pointer gestures

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#647]
Hi,
again I am trying to pin down approaches for testing, this time for 2.5.1 Complex pointer gestures. The issues:

- touchstart events not testable in desktop browsers (?) - I am not sure if and how a touch simulation bookmarklet like http://louisremi.github.io/touch-devtool/ or the Visual Event bookmarklet http://www.sprymedia.co.uk/article/Visual+Event+2 would work in practice for detecting the presence of complex touch gestures in a desktop testing environment.
- I have come across implementations with forking behaviour according to OS/UA (desktop site has alternative single point activation buttons for slider, mobile UA hasn't). See, for example,  https://www.zalando.de/get-the-look-damen/
- Multi-point gestures in web content (any examples, anyone?) would only be testable on mobile devices

The background of this search is that we have evaluators with motor impairments who simply cannot use mobile devices. So I am trying to identify valid testing approaches on the desktop (and identify tools that can guide or even replace monkey testing (tapping / swiping all over the interface to identify complex gestures).

**Comment by patrickhlauke** [comment]
> touchstart events not testable in desktop browsers

Developer tools in some browser have a device emulation (Chrome) / responsive design mode (Firefox) that allows for emulation of single pointer touchscreen interactions (no such option is available in IE11/Edge/Safari though).

> I have come across implementations with forking behaviour according to OS/UA

It will depend how sites try to detect this. If it's just down to UA sniffing, then you can still use devtools to send a fake/custom UA string. If they do naive tests to see if it's a "touch" device, it might - depending how they do their naive test - be possible to trigger/fool the site by using touch emulation as above. Worst case, you may need to use an actual emulator (Android SDK, XCode) or a related service like BrowserStack (where real mobile devices are used, but the site translates your mouse interactions into actual touch interactions - though not sure how accessible BrowserStack is overall).

> Multi-point gestures in web content (any examples, anyone?) would only be testable on mobile devices

Yes, that's the trickiest part which can't easily be tested/emulated at the moment, to my knowledge.

As in the other thread https://github.com/w3c/wcag/issues/643 it could also be an idea to very initially look for telltale signs in JavaScript - such as any `if ('ontouchstart' in window) { ... }` or similar naive checks for "mobile", and in general for any event handlers for `touchstart`, `touchmove`, `touchend`, `pointerdown`, `pointermove`, `pointerup`

**Comment by fstrr** [comment]
This discussion seems to be inactive so I'm closing it. If it needs to be re-opened, please do.
